driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
I've noticed that for this to work, you need to pass in the full name of the selected value.  I.e. if doing:

    select token(i) from test.foo where token i = 1

you would need to call ```row.getToken("token(i)");```, is that the intent of this?  Should we change getToken(name) to implicitly add 'token(" + name + ")" or is it fine the way it is?  I could see it be misleading either way, so adding a note to the javadoc would be good.